File: src/main/java/ninja/hudy/infosmog/model/CountyRegion.java
package ninja.hudy.infosmog.model;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import java.util.Set;
import java.util.stream.Collectors;

@Entity
@DiscriminatorValue(RegionType.Values.COUNTY)
public class CountyRegion extends Region {

    public Set<CityRegion> getCities() {
        return getNestedRegions().stream()
                .filter(region -> RegionType.CITY
                        .toString()
                        .equals(region.getType()))
                .map(region -> (CityRegion) region)
                .collect(Collectors.toSet());
    }

}

File: src/main/kotlin/pl/org/smok/infosmog/model/CountyRegion.kt
package pl.org.smok.infosmog.model

import javax.persistence.DiscriminatorValue
import javax.persistence.Entity
import java.util.stream.Collectors

@Entity
@DiscriminatorValue(RegionType.Values.COUNTY)
class CountyRegion : Region() {

    val cities: Set<CityRegion>
        get() = nestedRegions
                .filter { RegionType.CITY.toString() == it.type }
                .map { it as CityRegion }
                .distinct()
                .toSet()

}
